test/system/src/test/java/io/pravega/test/system/AbstractReadWriteTest.java

```diff
@@ -552,6 +583,127 @@ private EventWriterConfig buildWriterConfig() {
+        // Creating 5 threads to update KVP table entry
+        List<CompletableFuture<Void>> writerList = new ArrayList<>();
+        for (Integer i = 1; i <= kvpCount; i++) {
+            String value = convertStringToBinary("Hello World");
+            Integer keyId = i;
+            writerList.add(writeKVP(scope, kvtName, kf, value, keyId, keyValueTableFactory));
+        }
+        Futures.allOf(writerList);
+        /*if (!Futures.await(Futures.allOf(writerList))) {
+            log.error("Writers stopped with exceptions");
+        }*/
+    }
+
+    private CompletableFuture<Void> writeKVP(String scope, String kvtName, String kf,
+                                     String value, Integer keyId, KeyValueTableFactory keyValueTableFactory) {
+        return CompletableFuture.runAsync(() -> {
+            try {
+                KeyValueTableInfo kvt = new KeyValueTableInfo(scope, kvtName);
+                keyValueTable = keyValueTableFactory.forKeyValueTable(kvt.getKeyValueTableName(), KEY_SERIALIZER, VALUE_SERIALIZER, KeyValueTableClientConfiguration.builder().build());
+                log.info("Start creating KVP id:{} in KVT {}", keyId, kvtName);
+                // Create KVP
+                Version insertKvp = keyValueTable.put(kf, keyId, value).join();
+                // get KVP and verify the insert value length
+                Assert.assertEquals("Value length is not matching", value.length(),
+                        keyValueTable.get(kf, keyId).join().getValue().length());
+                // Call API to update kvp table entry
+                updateKVPEntry(scope, kvtName, kf, value, keyId, keyValueTableFactory, testState.stopKvpWrite);
+                //updateKVPEntry(scope, kvtName, kf, value, keyId, keyValueTableFactory);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }, executorService);
+    }
+
+    void readFromKVP(String scope, String kvtName, String kf, int kvpCount, KeyValueTableFactory keyValueTableFactory) {
+        Exceptions.handleInterrupted(() -> Thread.sleep(1000));
+        log.info("Start KVP read operation of KVT {} and scope {}", kvtName, scope);
+        // Creating 5 threads to update KVP table entry
+        List<CompletableFuture<Void>> readerList = new ArrayList<>();
+        for (Integer i = 1; i <= kvpCount; i++) {
+            Integer keyId = i;
+            log.info("Start reading from KVP id:{} of KVT {} of scope {}", keyId, kvtName, scope);
+            readerList.add(readKVP(scope, kvtName, kf, keyId, keyValueTableFactory, testState.stopKvpRead));
+        }
+        Futures.allOf(readerList);
+    }
+
+    private CompletableFuture<Void> readKVP(String scope, String kvtName, String kf, Integer keyId,
+                                    KeyValueTableFactory keyValueTableFactory, AtomicBoolean stopKvpReadFlag) {
+        return CompletableFuture.runAsync(() -> {
```

Comment:
All the same comments in `writeKVP` apply here.

Reply:
resolved